### PR TITLE
THIS IS NOT ACTUALLY FROM MASTER

### PR DESCRIPTION
**Pull request by LukeWood**
this pull request isn't actually coming from master - i'm working on many different branches on the repository located @ https://github.com/LukeWood/bullet.io

Name: Luke Wood
SMU ID: 46802422

### CODE REVIEW

File: bullet/lib/bullet/impl.ex [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    Map.update!(bullet, :y, fn y -> y + @bullet_speed end)
+  end
+
+  def distance(x1, x2, y1, y2) do
+    :math.sqrt(
+      :erlang.abs(:math.pow(x1-x2, 2) - :math.pow(y1-y2, 2))
+    )
+  end
+
+  defp did_collide(x1, x2, y1, y2, radius) do
```
Comment (pragdave) [inline bullet/lib/bullet/impl.ex:24]: maybe `did_collide?`

File: bullet/lib/bullet/supervisor.ex [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  use Supervisor
+
+  def start(_opts, _args) do
+    Supervisor.start_link(__MODULE__, :ok, name: __MODULE__)
+  end
+
+  def dx do
+    5 + Bullet.radius + 1
+  end
+
```
Comment (pragdave) [inline bullet/lib/bullet/supervisor.ex:11]: why are these transform functions in the supervisor?

File: clock/lib/clock.ex [hunk 3]
```diff
@@ -0,0 +1,21 @@
+defmodule Clock do
+  use Observable
+
+  @clock_speed 50
+
```
Comment (pragdave) [inline clock/lib/clock.ex:5]: I hate to diss an entire module, but would `:timer.send_interval` work here?

File: player/lib/player/impl.ex [hunk 4]
```diff
@@ -0,0 +1,52 @@
+defmodule Player.Impl do
+
+  @player_speed 1
+  defp move_player(player = %{moving: false}) do
+    player
+  end
```
Comment (pragdave) [inline player/lib/player/impl.ex:6]: There seem to be at least three versions of move up/down/left/right on tick. Could this be extracted out into a common library? Protocols might help (although you might not need them...)

**Comment by pragdave** [comment]
looking great—my only comments are fine tuning.

**Comment by LukeWood** [comment]
I've been struggling a bit with the dependency tree - is it reasonable to have a "Constants" module that has all of the constants like "player_speed" in them so that bullet can refer to player_speed without importing the player module

File: clock/lib/clock.ex (hunk 3 above)
Comment (LukeWood) [reply clock/lib/clock.ex:5]: Where do you think this code should live?  Still in a clock module or just in some sort of main module?

**Pull request closed without merging**